### PR TITLE
Add example OPA policy for blocking managed resources

### PR DESCRIPTION
**Pull request by nabuskey**
### What does this PR do?

Adds an example policy for blocking managed resources. 

### Motivation

Current OPA example only blocks compositions based on claim name space, but there is no example for blocking creation of managed resources. 

### More

- [x] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)

- [x] Yes, I have added a new example under [examples](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/examples) to support my PR

- [x] Yes, I have updated the [docs](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/doc) for this feature

- [x] Yes, I have linked to an issue or feature request (applicable to PRs that solves a bug or a feature request)

**Note**:
 - Not all the PRs require examples and docs
 - We prefer small, well tested pull requests. Please ensure your pull requests are self-contained, and commits are squashed

### For Moderators
- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: examples/gatekeeper/managed-resource/samples/allowed.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  - finalizer.managedresource.crossplane.io
+  generateName: standard-object-storage-9mfmk-
+  generation: 2
+  labels:
+    crossplane.io/claim-name: standard-object-storage
+    crossplane.io/claim-namespace: application4
+    crossplane.io/composite: standard-object-storage-9mfmk
+  name: standard-object-storage-9mfmk-xgqv8
+  ownerReferences:
+  - apiVersion: crossplane.adobe.io/v1alpha1
```
Comment (nimakaviani) [inline examples/gatekeeper/managed-resource/samples/allowed.yaml:20]: can we change this to a non-specific reference? ie something that doesnt tie it to adobe?

File: examples/gatekeeper/managed-resource/template.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        }
+
+        violation[{"msg": msg}] {
+          not input.review.object.metadata.ownerReferences
+          msg := "ownerReferences field not present"
+        }
+
+        violation[{"msg": msg}] {
+          ownerRef = input.review.object.metadata.ownerReferences[0]
+          not re_match("^crossplane.adobe.io", ownerRef.apiVersion)
```
Comment (nimakaviani) [inline examples/gatekeeper/managed-resource/template.yaml:42]: needs to match whatever we change it to from the earlier comment

File: examples/gatekeeper/managed-resource/samples/constraint.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+apiVersion: constraints.gatekeeper.sh/v1beta1
+kind: RequireComposition
+metadata:
+  name: crossplane-provider-config-ns
+spec:
+  match:
+    scope: "Cluster"
+    kinds:
+      - apiGroups: ["*"] # not possible to partial match. e.g. *.aws.crossplane.io does not work.
```
Comment (nimakaviani) [inline examples/gatekeeper/managed-resource/samples/constraint.yaml:9]: I am curious, is the lack of a partial match a limitation with gatekeeper?

File: examples/gatekeeper/managed-resource/samples/invalid_ownere_ref_disallowed.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+apiVersion: s3.aws.crossplane.io/v1beta1
+kind: Bucket
+metadata:
+  annotations:
+    crossplane.io/composition-resource-name: s3-bucket
+    crossplane.io/external-create-pending: "2022-09-29T16:36:09Z"
+    crossplane.io/external-create-succeeded: "2022-09-29T16:36:10Z"
+    crossplane.io/external-name: standard-object-storage-9mfmk-xgqv8
+  creationTimestamp: "2022-09-29T16:36:08Z"
```
Comment (nimakaviani) [inline examples/gatekeeper/managed-resource/samples/invalid_ownere_ref_disallowed.yaml:9]: we can probably get rid of these extra post-submission labels to make it a bit cleaner. not a big deal. just for better readability

File: examples/gatekeeper/managed-resource/samples/constraint.yaml (hunk 3 above)
Comment (nabuskey) [reply examples/gatekeeper/managed-resource/samples/constraint.yaml:9]: It is a limitation with Gatekeeper.

**Review by nimakaviani — APPROVED** [review]
LGTM

**Pull request merged**